### PR TITLE
Added tests for Tree component

### PR DESCRIPTION
**Pull request by krolwofficial**

### CODE REVIEW

File: src/alto-ui/Tree/components/TreeItem/__tests__/TreeItem.test.js [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    return ChevronRight;
+});
+
+describe('TreeItem', () => {
+    beforeEach(() => jest.resetAllMocks());
+
+    const getWrapper = (defaultProps, props) => mount(<TreeItem {...defaultProps} {...props} />);
+
+    describe('it', () => {
+        const defaultProps = {
```
Comment (barteksiudak) [inline src/alto-ui/Tree/components/TreeItem/__tests__/TreeItem.test.js]: This test is incomplete!
**first you should test `isSelected` prop**
* if `selected === item`
* if `selected !== item`
* if `selected === getKey(item, keyField)`
* if `selected !== getKey(item, keyField)`
* `getKey` toHaveBeenCalledWith(...)

**Icon prop**
On renderIcon what is heppening.

**ButtonOrLink prop
if href is a function or not

**test for hasChildren prop**

**href prop - toHaveBeenCalledTimes & toHaveBeenCalledWith**

**Review by barteksiudak — CHANGES_REQUESTED** [review]
Where are tests for TreeItemContainer ?
Where are `;` ?

File: src/alto-ui/Tree/__tests__/Tree.test.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+import { getKey } from './../helpers';
+
+jest.mock('./../helpers', () => ({
+	getKey: jest.fn(),
+}));
+
+describe('Tree', () => {
+    beforeEach(() => {
+        jest.resetAllMocks()
```
[the remark this answers is not in the document]
Comment (krolwofficial) [reply src/alto-ui/Tree/__tests__/Tree.test.js]: done with `getKey.mockClear();`

File: src/alto-ui/Tree/__tests__/Tree.test.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        getKey.mockReturnValue('test')
```
[the remark this answers is not in the document]
Comment (krolwofficial) [reply src/alto-ui/Tree/__tests__/Tree.test.js]: refactored to `getKey: jest.fn(({ key }) => key),`

File: src/alto-ui/Tree/__tests__/Tree.test.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    });
+
+    const defaultProps = {
+        id: 'id',
+        items: [{}],
```
[the remark this answers is not in the document]
Comment (krolwofficial) [reply src/alto-ui/Tree/__tests__/Tree.test.js]: Refactored

File: src/alto-ui/Tree/__tests__/Tree.test.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    it('should render without error',() => {
+        const wrapper = getWrapper({});
+        
+        expect(wrapper.find('.Tree').exists()).toBe(true);
+    })
+
+    describe('getKey func', () => {
+        it('should pass formated value to TreeItem id value',() => {
+    
+            const wrapper = getWrapper({});
```
[the remark this answers is not in the document]
Comment (krolwofficial) [reply src/alto-ui/Tree/__tests__/Tree.test.js]: Refactored everywhere

File: src/alto-ui/Tree/__tests__/Tree.test.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+                    
+            expect(wrapper.find('#id__item--test-0').key()).toEqual('test')
+        })
+    
+        it('should pass formated value to TreeItem key value',() => {    
+            const wrapper = getWrapper({});
+                    
+            expect(wrapper.find('#id__item--test-0').key()).toEqual('test')
+        })
+    })
```
[the remark this answers is not in the document]
Comment (krolwofficial) [reply src/alto-ui/Tree/__tests__/Tree.test.js]: Done
```
describe('getKey func', () => {
    it('should be called 4 times', () => {
```

File: src/alto-ui/Tree/components/TreeItem/TreeItem.js [hunk 7]
```diff
@@ -107,7 +107,7 @@ TreeItem.propTypes = {
     fetching: PropTypes.bool,
     children: PropTypes.any,
   }),
-  selected: PropTypes.string,
+  selected: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
```
[the remark this answers is not in the document]
Comment (krolwofficial) [reply src/alto-ui/Tree/components/TreeItem/TreeItem.js]: Yes, we figured it out with Damian

File: src/alto-ui/Tree/components/TreeItem/__tests__/TreeItem.test.js [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+import TreeItem from '../TreeItem'
+
+import { getKey } from '../../../helpers';
+
+jest.mock('../../../helpers', () => ({
+	getKey: jest.fn(),
+}));
+
+jest.mock('../../../../Icons/ChevronRight', () =>  {
+    const ChevronRight = () => <div className="ChevronRight" />;
```
[the remark this answers is not in the document]
Comment (krolwofficial) [reply src/alto-ui/Tree/components/TreeItem/__tests__/TreeItem.test.js]: Refactored

File: src/alto-ui/Tree/components/TreeItem/__tests__/TreeItem.test.js [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+    return ChevronRight;
+});
+
+describe('TreeItem', () => {
+    beforeEach(() => jest.resetAllMocks());
+
+    const getWrapper = (defaultProps, props) => mount(<TreeItem {...defaultProps} {...props} />);
+
+    describe('it', () => {
```
[the remark this answers is not in the document]
Comment (krolwofficial) [reply src/alto-ui/Tree/components/TreeItem/__tests__/TreeItem.test.js]: Removed

File: src/alto-ui/Tree/components/TreeItem/__tests__/TreeItem.test.js [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+
+jest.mock('../../../../Icons/ChevronRight', () =>  {
+    const ChevronRight = () => <div className="ChevronRight" />;
+    return ChevronRight;
+});
+
+describe('TreeItem', () => {
+    beforeEach(() => jest.resetAllMocks());
+
+    const getWrapper = (defaultProps, props) => mount(<TreeItem {...defaultProps} {...props} />);
```
[the remark this answers is not in the document]
Comment (krolwofficial) [reply src/alto-ui/Tree/components/TreeItem/__tests__/TreeItem.test.js]: Refactored to `const getWrapper = props => mount(<TreeItem {...defaultProps} {...props} />);`

File: src/alto-ui/Tree/components/TreeItem/__tests__/TreeItem.test.js (hunk 1 above)
Comment (krolwofficial) [reply src/alto-ui/Tree/components/TreeItem/__tests__/TreeItem.test.js]: 1. Added tests for `isSelected `prop:
```
- it('should render with outline prop', () => {
- it('should render without outline prop', () => {
- it('getKey toHaveBeenCalledWith(...)', () => {
```

2. On renderIcon prop func the Icon is rendering. I have this prop in default props:
`renderIcon: jest.fn().mockReturnValue(() => <div />),`
If the renderIcon prop is falsy the Icon should render null:
```
it('should render null', () => {
      expect(
        getCurrentWrapper({ renderIcon: null })
          .find('.TreeItem__icon')
          .exists()
      ).toBe(false);
      expect(
        getCurrentWrapper({ renderIcon: undefined })
          .find('.TreeItem__icon')
          .exists()
      ).toBe(false);
    });
```

3. `it('should render as Link element', () => {` <- if href is a function
`it('should render as button element', () => {` <- if href is not a function

4. I have tested hasChildren prop in `it('should render TreeItem__title--final', () => {` assertion

5. I added the assertion for href prop
`it('should render with href attribute', () => {`

**Pull request merged**